src/main/java/com/comphenix/protocol/wrappers/WrappedLevelChunkData.java

```diff
@@ -0,0 +1,482 @@
+            HEIGHTMAPS_ACCESSOR.set(handle, NbtFactory.fromBase(heightmapsTag).getHandle());
+        }
+
+        /**
+         * The actual structural data of this chunk as bytes.
+         *
+         * @return a byte array containing the chunks structural data.
+         */
+        public ByteBuf getBuffer() {
+            return Unpooled.wrappedBuffer((byte[]) BUFFER_ACCESSOR.get(handle));
+        }
+
+        /**
+         * Sets the structural data of this chunk using a {@link ByteBuf}.
+         *
+         * @param buffer the new buffer.
+         */
+        public void setBuffer(ByteBuf buffer) {
+            byte[] extracted = new byte[buffer.readableBytes()];
+
+            buffer.readBytes(extracted);
+
+            setBuffer(extracted);
+        }
+
+        /**
+         * Sets the structural data of this chunk.
+         *
+         * @param buffer the new buffer.
+         */
+        public void setBuffer(byte[] buffer) {
+            BUFFER_ACCESSOR.set(handle, buffer);
+        }
+
+        /**
+         * All block entities of this chunk. Supports removal and other edits.
+         *
+         * @return a mutable list containing {@link  BlockEntityInfo}
+         */
+        public List<BlockEntityInfo> getBlockEntityInfo() {
+            //noinspection StaticPseudoFunctionalStyleMethod
+            return Lists.transform((List<?>) BLOCK_ENTITIES_DATA_ACCESSOR.get(handle), BlockEntityInfo::new);
+        }
+
+        private static ConstructorAccessor levelChunkPacketDataConstructor;
+
+        /**
+         * Creates a new wrapper using predefined values.
+         *
+         * @param heightmapsTag   the heightmaps tag
+         * @param buffer          the buffer
+         * @param blockEntityInfo a list of wrapped block entities
+         * @return a newly created wrapper
+         */
+        public static ChunkData fromValues(NbtCompound heightmapsTag, byte[] buffer, List<BlockEntityInfo> blockEntityInfo) {
+            if (levelChunkPacketDataConstructor == null) {
+                levelChunkPacketDataConstructor = Accessors.getConstructorAccessor(HANDLE_TYPE, MinecraftReflection.getPacketDataSerializerClass(), int.class, int.class);
+            }
+
+            ConstructorAccessor trickySerializer = StructureCache.getTrickDataSerializerOrNull();
+
+            if (trickySerializer == null) {
+                throw new UnsupportedOperationException("TrickySerializer is not supported");
+            }
+
+            Object instance = levelChunkPacketDataConstructor.invoke(trickySerializer.invoke(new ZeroBuffer()), 0, 0);
+
+            BUFFER_ACCESSOR.set(instance, buffer);
+            HEIGHTMAPS_ACCESSOR.set(instance, NbtFactory.fromBase(heightmapsTag).getHandle());
+
+            for (BlockEntityInfo entityInfo : blockEntityInfo) {
+                //noinspection unchecked
+                ((List) BLOCK_ENTITIES_DATA_ACCESSOR.get(instance)).add(entityInfo.getHandle());
```

Comment:
Thats not a good idea, just put the list into a local field and then add all elements (accessor get is always reflective which shouldn't be heavy, but noticeable depending on the amount of block entities) 😄